gio/src/subclass/icon.rs

```diff
@@ -0,0 +1,101 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::Icon;
+use glib::subclass::prelude::*;
+use glib::translate::*;
+use glib::{Cast, IsA};
+
+pub trait IconImpl: ObjectImpl {
+    fn hash(&self, icon: &Self::Type) -> u32;
+    fn equal(&self, icon: &Self::Type, other_icon: &Icon) -> bool;
+    fn to_tokens(&self, icon: &Self::Type) -> Option<(Vec<String>, i32)>;
```

Comment:
Is that `i32` an icon size? I think we have an enum for that?

Reply:
Ah no, it's a version Nevermind

Reply:
the i32 here is the "version" that should be passed when using from_tokens